src/platform/site-wide/header/components/App/index.js

```diff
@@ -1,43 +1,116 @@
 import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import recordEvent from '~/platform/monitoring/record-event';
-import Header from '../Header';
-import { hideLegacyHeader, showLegacyHeader } from '../../helpers';
+import debounce from 'platform/utilities/data/debounce';
+import { updateLayoutHeaderType } from 'platform/site-wide/layout/actions';
+import { useSelector, useDispatch } from 'react-redux';
+import MobileHeader from '../Header';
+import {
+  hideLegacyHeader,
+  showLegacyHeader,
+  toggleMinimalHeader,
+} from '../../helpers';
 
 const MOBILE_BREAKPOINT_PX = 768;
 
-export const App = ({ megaMenuData, show, showMegaMenu, showNavLogin }) => {
+function determineHeaderState(show, showMinimalHeader, path, isDesktop) {
+  if (!show) {
+    return 'none';
+  }
+  if (
+    showMinimalHeader &&
+    (typeof showMinimalHeader === 'function'
+      ? showMinimalHeader(path)
+      : showMinimalHeader)
+  ) {
+    return 'minimal';
+  }
+
+  if (!isDesktop) {
+    return 'mobile';
+  }
+
+  return 'default';
+}
+
+function setStaticHeaderDisplay(
+  show,
+  isMinimalHeaderApplicable,
+  headerState,
+  isDesktop,
+) {
+  if (show) {
+    if (isMinimalHeaderApplicable) {
+      toggleMinimalHeader(headerState === 'minimal');
+    }
+    if (isDesktop) {
+      showLegacyHeader();
+    } else {
+      hideLegacyHeader();
+    }
+  }
+  // else everything is already hidden by default
+}
+
+export const App = ({
+  megaMenuData,
+  show,
+  showMegaMenu,
+  showNavLogin,
+  showMinimalHeader,
+}) => {
+  const dispatch = useDispatch();
+  const path = useSelector(state => state?.navigation?.route?.path);
+  const [headerState, setHeaderState] = useState(null);
   const [isDesktop, setIsDesktop] = useState(
     window.innerWidth >= MOBILE_BREAKPOINT_PX,
   );
 
-  const deriveIsDesktop = () =>
-    setIsDesktop(window.innerWidth >= MOBILE_BREAKPOINT_PX);
-
   useEffect(() => {
+    const deriveIsDesktop = () =>
+      setIsDesktop(window.innerWidth >= MOBILE_BREAKPOINT_PX);
+
+    const onResize = debounce(100, deriveIsDesktop);
+
     recordEvent({
       event: 'phased-roll-out-enabled',
       'product-description': 'Header V2',
     });
 
-    window.addEventListener('resize', deriveIsDesktop);
+    window.addEventListener('resize', onResize);
 
-    return () => window.removeEventListener('resize', deriveIsDesktop);
+    return () => window.removeEventListener('resize', onResize);
   }, []);
 
-  if (!show) {
-    return null;
-  }
+  useEffect(
+    () => {
+      const newHeaderState = determineHeaderState(
+        show,
+        showMinimalHeader,
+        path,
+        isDesktop,
+      );
+
+      if (headerState !== newHeaderState) {
+        setStaticHeaderDisplay(
```

Comment:
Before, the querySelectors for hiding/showing static DOM was done in the normal component render flow, but I moved it to a useEffect, because that is React's recommendation.

The result is that the code to display legacy header will render 1 paint cycle later than it previously did. Just bringing this up in case anyone anyone has any concerns - but perceivably it felt the same, but you can notice the slight flicker of footer rendering before the header.